src/core/net/ip6.cpp

```diff
@@ -328,6 +333,24 @@ void Ip6::ProcessReceiveCallback(Message &aMessage)
 
     VerifyOrExit(sReceiveIp6DatagramCallback != NULL, ;);
 
+    if (sFilterControlTraffic)
+    {
+        Header ip6;
+        aMessage.Read(0, sizeof(ip6), &ip6);
+
+        // do not pass messages sent to/from an RLOC
+        VerifyOrExit(!ip6.GetSource().IsRoutingLocator() && !ip6.GetDestination().IsRoutingLocator(), ;);
+
+        if (ip6.GetNextHeader() == kProtoUdp)
```

Comment:
Shouldn't this only be filtered for the ML-EID address? What if there are other addresses that end up using that same port randomly?